api/types/desktop.go

```diff
@@ -127,6 +127,198 @@ func (s *WindowsDesktopServiceV3) MatchSearch(values []string) bool {
 	return MatchSearch(fieldVals, values, nil)
 }
 
+// DynamicWindowsDesktop represents a Windows desktop host that is automatically discovered by Windows Desktop Service.
+type DynamicWindowsDesktop interface {
+	// ResourceWithLabels provides common resource methods.
+	ResourceWithLabels
+	// GetAddr returns the network address of this host.
+	GetAddr() string
+	// GetDomain returns the ActiveDirectory domain of this host.
+	GetDomain() string
+	// NonAD checks whether this is a standalone host that
+	// is not joined to an Active Directory domain.
+	NonAD() bool
```

Comment:
I'd name it `IsStandalone` or `IsAD` instead of returning a "negated" name.

Reply:
It's for consistency between `DynamicWindowsDesktop` and `WindowsDesktop`, I'll leave it as is